app/src/main/java/com/tohami/newsapi/ui/news/NewsActivity.java

```diff
@@ -0,0 +1,108 @@
+package com.tohami.newsapi.ui.news;
+
+import android.content.res.Configuration;
+import android.os.Bundle;
+import android.os.PersistableBundle;
+import android.view.View;
+import android.widget.Toast;
+
+import androidx.appcompat.app.ActionBarDrawerToggle;
+import androidx.appcompat.widget.Toolbar;
+import androidx.drawerlayout.widget.DrawerLayout;
+import androidx.navigation.NavController;
+import androidx.navigation.Navigation;
+
+import com.google.android.material.navigation.NavigationView;
+import com.tohami.newsapi.R;
+import com.tohami.newsapi.base.BaseActivity;
+
+public class NewsActivity extends BaseActivity {
+    private ActionBarDrawerToggle mDrawerToggle;
+    private DrawerLayout drawerLayout;
+    private NavController navController;
+    private NavigationView navigationView;
+
+    @Override
+    protected void onCreate(Bundle savedInstanceState) {
+        super.onCreate(savedInstanceState);
+    }
+
+    @Override
+    protected void initializeViews() {
+        Toolbar appbar = findViewById(R.id.toolbar);
+        drawerLayout = findViewById(R.id.drawer_layout);
+        navigationView = findViewById(R.id.nvView);
+        setToolbar(appbar, R.color.colorPrimary, getString(R.string.app_name), true, false);
+        navController = Navigation.findNavController(this, R.id.main_nav_container);
+    }
+
+    private void setUpNavDrawerListeners() {
+        mDrawerToggle = new ActionBarDrawerToggle(this, drawerLayout, myToolbar, R.string.app_name, R.string.app_name) {
```

Comment:
Any callbacks should be defined outside of main methods and only to be passed as parameters

Reply:
need discuss